src/rust/engine/src/nodes.rs

```diff
@@ -720,6 +748,33 @@ impl SelectProjection {
   }
 }
 
+///
+/// A Node that represents executing a process.
+///
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ExecuteProcess(process_executor::ExecuteProcessRequest);
+
+#[derive(Clone, Debug)]
+pub struct ProcessResult(process_executor::ExecuteProcessResult);
+
+impl Node for ExecuteProcess {
+  type Output = ProcessResult;
+
+  fn run(self, _: Context) -> NodeFuture<ProcessResult> {
+    let request = self.0.clone();
+    future::ok(ProcessResult(
+      process_executor::local::run_command_locally(request)
```

Comment:
Remind me... Why do we need two calls to `run_command_locally` in this file?

Reply:
We shouldn't... the explicit call in `fn gen_nodes` should be replaced with a request for this `Node` implementation.

I have a plan to refactor the python/rust interaction for process execution, so please don't spend any more time polishing the python/rust boundary than you need to to get the tests passing.